test/controllers/users_controller_test.rb

```diff
@@ -402,6 +402,16 @@ class UsersControllerTest < ActionController::TestCase
     assert_equal user.mail, mail.to[0]
   end
 
+  test "test email should be delivered even when user has read only role" do
+    user = users(:one)
+    user.roles = [Role.find_by_name('Viewer')]
+    User.current = user
+    put :test_mail, { :id => user.id, :user => {:login => user.login} }, set_session_user
```

Comment:
`set_session_user`, when not passed a parameter, defaults to run the request as admin. You should user `set_session_user(user)` instead.